src/nvim/shada.c

```diff
@@ -1886,13 +1886,21 @@ static inline ShaDaWriteResult shada_read_when_writing(FileDescriptor *const sd_
           shada_free_shada_entry(&entry);
           break;
         }
-        if (wms->global_marks[idx].data.type == kSDItemMissing) {
+
+        PossiblyFreedShadaEntry *global_or_numbered_mark;
+        if (idx < 26) {
+          global_or_numbered_mark = &wms->global_marks[idx];
+        } else {
+          global_or_numbered_mark = &wms->numbered_marks[idx];
+        }
```

Comment:
This is a very short-lived variable so doesn't need an elaborate name.

```suggestion
        // Global or numbered mark.
        PossiblyFreedShadaEntry *mark = idx < 26 ? &wms->global_marks[idx] : &wms->numbered_marks[idx];
```

Reply:
Appended your suggestion, thanks!

Reply:
Oh I didn't see that you already applied them, my mistake.